src/cpu_shader/pathtag_reduce.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright 2023 The Vello authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use vello_encoding::{ConfigUniform, Monoid, PathMonoid};
+
+use crate::cpu_dispatch::CpuBinding;
+
+const WG_SIZE: usize = 256;
+
+fn pathtag_reduce_main(
+    n_wg: u32,
+    config: &ConfigUniform,
+    scene: &[u32],
+    reduced: &mut [PathMonoid],
+) {
+    let pathtag_base = config.layout.path_tag_base;
+    for i in 0..n_wg {
+        let mut m = PathMonoid::default();
+        for j in 0..WG_SIZE {
+            let tag = scene[(pathtag_base + i * WG_SIZE as u32) as usize + j];
+            m = m.combine(&PathMonoid::new(tag));
```

Comment:
<details>
<summary>Musings on bad and premature optimisations</summary>
I wonder if these operations could be reordered to improve pipelining. That is, since operation 2 depends on operation 1 (according to the CPU), could we start operation 3/4 whilst 1/2 are ongoing.
My (likely unfounded) intuition around CPU implementations is that loop steps depending on the previous iteration is slow, because the operation must fully complete before the next can begin.

Is there some way to signal to LLVM that these operations are order-invariant?
</details>